code/include/playfab/PlayFabClientInstanceApi.h

```diff
@@ -90,7 +91,7 @@ namespace PlayFab
         void GetPaymentToken(ClientModels::GetPaymentTokenRequest& request, const ProcessApiCallback<ClientModels::GetPaymentTokenResult> callback, const ErrorCallback errorCallback = nullptr, void* customData = nullptr);
         void GetPhotonAuthenticationToken(ClientModels::GetPhotonAuthenticationTokenRequest& request, const ProcessApiCallback<ClientModels::GetPhotonAuthenticationTokenResult> callback, const ErrorCallback errorCallback = nullptr, void* customData = nullptr);
         void GetPlayerCombinedInfo(ClientModels::GetPlayerCombinedInfoRequest& request, const ProcessApiCallback<ClientModels::GetPlayerCombinedInfoResult> callback, const ErrorCallback errorCallback = nullptr, void* customData = nullptr);
-        void GetPlayerProfile(ClientModels::GetPlayerProfileRequest& request, const ProcessApiCallback<ClientModels::GetPlayerProfileResult> callback, const ErrorCallback errorCallback = nullptr, void* customData = nullptr);
+        void GetPlayerProfile(const PlayFabGetPlayerProfileRequest& request, const ProcessApiCallback<ClientModels::GetPlayerProfileResult> callback, const TaskQueue& queue, const ErrorCallback errorCallback = nullptr, void* customData = nullptr);
```

Comment:
I do think the requests should be const, and the changed lines isn't really a concern since this is for a new SDK anyhow :)

I think it makes the most sense to specify a queue for each API rather than for the entire API set.  By design, each XAsync API needs to take an XAsyncBlock parameter - its used to track the async call, get the result, cancel, etc.  That XAsyncBlock has a queue field, so if we say we are using a more "global" queue for the whole API set, we would effectively be ignoring the queue passed as part of the XAsyncBlock parameter.

This is slightly more tricky.  There might be a way to design this a bit differently, but right now the request object is a public "C" struct and thus doesn't inherit from PlayFabBaseRequest.  Will have to think about this a little more. 

---
In reply to: [590838325](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590838325) [](ancestors = 590838325)